### ISSUE TITLE
Kernel 4.19.79-1 breaks system

### ISSUE THREAD

**Issue by michaelschefczyk** [QubesOS/qubes-issues#5472]
Dear All,

When upgrading dom0, Kernel 4.19.79-1 was installed to my recent R4 QubesOS. That did break my system (Supermicro X11SCV-Q). Sys-net will not start as it is unable to reset the PCI device. However, I am unable to change VM setings or even open the qube manager.  Therefore, there is no network connectivty and nothing starts. It would be nice to at least be able to backup VMs before reinstalling.

What might help is to start with a previous kernel - 4.19.67-1 would be good. As the system is booting EFI, I am unfortunately unable to do this for lack of documentation and/or EFI-kowhow. Is it possible to open a grub like menu? If so, how? I did modify /boot/efi/EFI/qubes/xen.cfg according to https://www.qubes-os.org/doc/software-update-dom0/ However, the system has no /etc/default/grub and the like and no /usr/sbin/brub2-mkconfig.

Thanks & Regards,

Michael Schefczyk

**Comment by marmarek** [comment]
> I did modify /boot/efi/EFI/qubes/xen.cfg according to https://www.qubes-os.org/doc/software-update-dom0/

It is enough on EFI system. The other section (grub) is about legacy boot.

> However, I am unable to change VM setings or even open the qube manager.

What error you get here? "unable to reset the PCI device" should not prevent you from opening settings. If it's just silence, try opening terminal in dom0 and running qube manager from there: `qubes-qube-manager`

**Comment by michaelschefczyk** [comment]
Thank you very much for your super fast response! I did notice that the computer was chashing and repooting when I tried to apply the kernel update. That did happen multiple times. After the first one or two reboots, network connectivity was still there - otherwise I could not have tried updating again.

Unfortunately, starting qubes-qube-manager from the terminal emulator does not work - with and without root. The error message is (unable to copy, typing may contain errors):

Traceback (most recent call last):
File "/bin/qubes-qube-manager", line 9, in <module>
load_entry_point('qubesmanager==4.0.40', 'console_scripts', 'qubes-qube-manager')()
File "/usr/lib/python3.5/site-packages/pkg_resources/__init_.py", line 541, in load _entry_point
return get_distribution(dist).loac_entry_point(group.name)
File "Usr/lib/python3.5/site-packages/pkg_resources/__init_.py", line 2574, in load_entry_point
raise ImportError("Entry point %r not found" % ((group, name),))
ImportError: Entry point ('console_scripts', 'qubes-qube-manager' not found

My intuition tells me that I am due to reinstall without backing up my VMs first. Is that correct?

**Comment by marmarek** [comment]
Looks like some missing files, probably interrupted installation or out of disk space. If the updates were downloaded properly before there may still be in the cache, you can try to finish installing them. Try `sudo dnf update` in dom0  (not `qubes-dom0-update`!) or `sudo dnf reinstall qubes-manager`. You can also verify packages status with `sudo rpm -Va` (will take some time) - especially look for `missing` entries about files in `/usr` (those in /var/lib/qubes/vm-templates` are expected to be missing).

You can also try to start just `sys-usb`, connect some USB disk and make the backup using `qvm-backup` command line tool.
While fixing this installation is probably possible, it may involve some manual file copying, depending on that's actually missing. Reinstallation is probably much easier.

**Comment by michaelschefczyk** [comment]
Thanks again! dnf update led to the installation of kernel 4.19.82. However, that did not solve the problem. dnf reinstall qubes-manager did not help either, because qubes-manager is not available (at least without network, it seems). qvm-start sys-usb does not work either, because I have two usb controllers of which one is assigned to sys-usb. Sys-usb will not start because of "unable to reset the PCI device". What might work is qvm-backup command line using the controller which is attached to dom0 - and which does allow mouse and keyboard access.

**Comment by andrewdavidwong** [comment]
It appears that this has turned out to be more of a support request than a bug report, since it is not clear that there is actually any bug involving kernel 4.19.79-1. Is this correct?

**Comment by michaelschefczyk** [comment]
I will report back once I will understand this. It might be a hardware compatibility issue. I have re-flashed the bios on the machine, removed hardware to ensure that there are no power issues, exchanged the nvme ssd to exclude fault of that as a course. Live OS on legacy boot do work on the machine. I am re-creating the Qubes installation medium and trying the new RC version and I plan also to test by installing either Debian or CentOS UEFI on the system if needed and then I will report back.

**Comment by michaelschefczyk** [comment]
This turns out to be not a support issue, but more likely a serious hardware compatiblity issue.

I use qubes on three systems. The VMs are almost identical and many of them were copied in between the sytems. The most frequently used (and thus most frequently updated) system is a Lenovo notbook. That has no issues whatsoever. The two other ones are less frequently used, identical desktop systems. They have a Supermicro X11SCV-Q motherboard, an i7 processor and a Samsung 970 Pro NVME SSD, a secondary USB controller plugged into the PCIe slot and a Samsung 850 Pro SATA SSD as an internal backup drive. Both systems failed at the identical update step in identical manners after working for more than a year without issues.

From dnf.log, the systems crashed during the following updates:

================================================================================
 Package                    Arch   Version              Repository         Size
================================================================================
Installing:
 kernel                     x86_64 1000:4.19.82-1.pvops.qubes
                                                        qubes-dom0-cached  49 M
 kernel-qubes-vm            x86_64 1000:4.19.82-1.pvops.qubes
                                                        qubes-dom0-cached  78 M
 linux-firmware-whence      noarch 20190923-102.1.fc25  qubes-dom0-cached  46 k
 socat                      x86_64 1.7.3.2-1.fc25       qubes-dom0-cached 291 k
Upgrading:
 anaconda-core              x86_64 1000:25.20.9-16.fc25 qubes-dom0-cached 1.5 M
 anaconda-gui               x86_64 1000:25.20.9-16.fc25 qubes-dom0-cached 386 k
 anaconda-tui               x86_64 1000:25.20.9-16.fc25 qubes-dom0-cached 186 k
 anaconda-widgets           x86_64 1000:25.20.9-16.fc25 qubes-dom0-cached 119 k
 device-mapper-persistent-data
                            x86_64 0.7.5-3.fc25         qubes-dom0-cached 435 k
 libvirt-client             x86_64 3.3.0-9.fc25         qubes-dom0-cached 287 k
 libvirt-daemon             x86_64 3.3.0-9.fc25         qubes-dom0-cached 660 k
 libvirt-daemon-driver-interface
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  55 k
 libvirt-daemon-driver-libxl
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached 151 k
 libvirt-daemon-driver-nodedev
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  54 k
 libvirt-daemon-driver-nwfilter
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  78 k
 libvirt-daemon-driver-secret
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  44 k
 libvirt-daemon-driver-storage
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached 6.2 k
 libvirt-daemon-driver-storage-core
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached 185 k
 libvirt-daemon-driver-storage-disk
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  15 k
 libvirt-daemon-driver-storage-gluster
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  16 k
 libvirt-daemon-driver-storage-iscsi
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  12 k
 libvirt-daemon-driver-storage-logical
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  16 k
 libvirt-daemon-driver-storage-mpath
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  11 k
 libvirt-daemon-driver-storage-scsi
                            x86_64 3.3.0-9.fc25         qubes-dom0-cached  12 k
 libvirt-daemon-driver-xen  x86_64 3.3.0-9.fc25         qubes-dom0-cached 131 k
 libvirt-daemon-xen         x86_64 3.3.0-9.fc25         qubes-dom0-cached 6.1 k
 libvirt-libs               x86_64 3.3.0-9.fc25         qubes-dom0-cached 3.8 M
 linux-firmware             noarch 20190923-102.1.fc25  qubes-dom0-cached  62 M
 python2-qubesadmin         noarch 4.0.28-1.fc25        qubes-dom0-cached 116 k
 python2-qubesimgconverter  x86_64 4.0.26-1.fc25        qubes-dom0-cached  26 k
 python3-blivet             noarch 2:2.1.11-6.fc25      qubes-dom0-cached 1.0 M
 python3-qubesadmin         noarch 4.0.28-1.fc25        qubes-dom0-cached 367 k
 python3-qubesdb            x86_64 4.0.12-1.fc25        qubes-dom0-cached  18 k
 python3-qubesimgconverter  x86_64 4.0.26-1.fc25        qubes-dom0-cached  27 k
 python3-xen                x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached  59 k
 qubes-core-admin-client    noarch 4.0.28-1.fc25        qubes-dom0-cached  61 k
 qubes-core-dom0            x86_64 4.0.48-1.fc25        qubes-dom0-cached 735 k
 qubes-core-dom0-linux      x86_64 4.0.21-1.fc25        qubes-dom0-cached  52 k
 qubes-core-dom0-linux-kernel-install
                            x86_64 4.0.21-1.fc25        qubes-dom0-cached  14 k
 qubes-db                   x86_64 4.0.12-1.fc25        qubes-dom0-cached  29 k
 qubes-db-dom0              x86_64 4.0.12-1.fc25        qubes-dom0-cached  12 k
 qubes-db-libs              x86_64 4.0.12-1.fc25        qubes-dom0-cached  17 k
 qubes-desktop-linux-common noarch 4.0.19-1.fc25        qubes-dom0-cached 137 k
 qubes-desktop-linux-manager
                            noarch 4.0.22-1.fc25        qubes-dom0-cached  66 k
 qubes-gpg-split-dom0       x86_64 2.0.41-1.fc25        qubes-dom0-cached  28 k
 qubes-libvchan-xen         x86_64 4.0.7-1.fc25         qubes-dom0-cached  19 k
 qubes-manager              noarch 4.0.40-1.fc25        qubes-dom0-cached 1.8 M
 qubes-menus                noarch 4.0.19-1.fc25        qubes-dom0-cached  50 k
 qubes-mgmt-salt-dom0-qvm   noarch 4.0.10-1.fc25        qubes-dom0-cached  48 k
 qubes-usb-proxy-dom0       noarch 1.0.22-1.fc25        qubes-dom0-cached  37 k
 qubes-utils                x86_64 4.0.26-1.fc25        qubes-dom0-cached  23 k
 qubes-utils-libs           x86_64 4.0.26-1.fc25        qubes-dom0-cached  27 k
 xen                        x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached  24 k
 xen-hvm                    x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached 7.3 M
 xen-hypervisor             x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached 6.2 M
 xen-libs                   x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached 520 k
 xen-licenses               x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached  52 k
 xen-runtime                x86_64 2001:4.8.5-10.fc25   qubes-dom0-cached 8.1 M
 xscreensaver-base          x86_64 1:5.43-1.fc25        qubes-dom0-cached 456 k
Removing:
 kernel                     x86_64 1000:4.14.74-1.pvops.qubes
                                                        @anaconda         239 M
 kernel-qubes-vm            x86_64 1000:4.14.74-1.pvops.qubes
                                                        @anaconda         272 M

Transaction Summary
================================================================================
Install   4 Packages
Upgrade  55 Packages
Remove    2 Packages

After the crash, the system wants to do the following updates:

================================================================================
 Package              Arch   Version                    Repository         Size
================================================================================
Installing:
 kernel               x86_64 1000:4.19.82-1.pvops.qubes qubes-dom0-cached  49 M
 kernel-qubes-vm      x86_64 1000:4.19.82-1.pvops.qubes qubes-dom0-cached  78 M
Upgrading:
 anaconda-gui         x86_64 1000:25.20.9-16.fc25       qubes-dom0-cached 386 k
 device-mapper-persistent-data
                      x86_64 0.7.5-3.fc25               qubes-dom0-cached 435 k
 libvirt-client       x86_64 3.3.0-9.fc25               qubes-dom0-cached 287 k
 python2-qubesimgconverter
                      x86_64 4.0.26-1.fc25              qubes-dom0-cached  26 k
 qubes-desktop-linux-manager
                      noarch 4.0.22-1.fc25              qubes-dom0-cached  66 k
 qubes-gpg-split-dom0 x86_64 2.0.41-1.fc25              qubes-dom0-cached  28 k
 qubes-menus          noarch 4.0.19-1.fc25              qubes-dom0-cached  50 k
 qubes-mgmt-salt-dom0-qvm
                      noarch 4.0.10-1.fc25              qubes-dom0-cached  48 k
 qubes-usb-proxy-dom0 noarch 1.0.22-1.fc25              qubes-dom0-cached  37 k
 xscreensaver-base    x86_64 1:5.43-1.fc25              qubes-dom0-cached 456 k
Removing:
 kernel               x86_64 1000:4.14.74-1.pvops.qubes @anaconda         239 M
 kernel-qubes-vm      x86_64 1000:4.14.74-1.pvops.qubes @anaconda         272 M

Transaction Summary
================================================================================
Install   2 Packages
Upgrade  10 Packages
Remove    2 Packages

That would fail again and in the end, one can upgrade kernel and kernel-qubes-vm to 4.19.82-1. One system was damaged on the way (my original question at the start of this post). After reinstalling from scratch, the above did happen to the fresh system similar to the other, not damaged system. The findings being identical on two independent systems (kept in two different cities) makes it unlikely that a hardware component (CPU, RAM ...) developed a failure spontaneously.

After that, the system will crash more frequently - which did never happen before. My impression is that the system will crash if too much data is written to the NVME SSD in a very brief period. I suspect that this might happen in two situations which I have observed:
a) The upgrades as shown here.
b) Resoring backups of larger VMs, i.e., larger than ca. 5 GB.

Is there anything one can do about this? It would be a pitty to find that the nice, compact desktop PCs based on the Supermicro X11SCV-Q motherboard are no longer useful after this update.

**Comment by xaki23** [comment]
this really sounds like "running out of space", in a way that is corrupting the install. (no, "df" can NOT rule that out.)
it could also (in parts) be bad usage of thin devices, like sparse-restoring a template over an existing template volume without blkdiscarding first.

i would recommend a clean test.
reinstall.
without altering install config (beyond the "pick lang/kbd/timezone/disk" basics).
without any kind of restore.
without any experimental/advanced config like multi-usb setup.
update dom0 + templates.
reboot.

this whole "system crashes during update" is a massive red flag already. 
thats a "should never happen".
and it is unlikely this crash-while-installing-updates is caused _by_ the updates that are not running yet.

**Comment by michaelschefczyk** [comment]
Having tried with two identical computers - unfortunately in two different cities, which is why it took so long - I determined, that this is a hardware issue purely on my side. The systems did have passive CPU cooling so far. This never created any issues over many months until this very update did seemingly overload both systems. I did install an active CPU heat sink in both systems and afterwards, they did survive the update. Sorry for taking time to explore other potential causes.
